Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Sharing/SocketerNetworkingService.cs

```diff
@@ -0,0 +1,188 @@
+        [SerializeField]
+        protected TimeSpan timeoutInterval = TimeSpan.Zero;
+
+        private bool runAsUser = false;
+        private Dictionary<string, string> playerIds = new Dictionary<string, string>();
+
+        /// <inheritdoc />
+        public event PlayerConnectedHandler PlayerConnected;
+
+        /// <inheritdoc />
+        public event PlayerDisconnectedHandler PlayerDisconnected;
+
+        /// <inheritdoc />
+        public event DataHandler DataReceived;
+
+        private void Start()
+        {
+#if UNITY_WSA && !UNITY_EDITOR
+            runAsUser = true;
+#elif UNITY_ANDROID || UNITY_IOS || UNITY_EDITOR
+            runAsUser = false;
+#endif
+        }
+
+        /// <inheritdoc />
+        public void Connect()
+        {
+            if (net == null)
+            {
+                Debug.LogError("TCPConnectionManager not set for SocketNetworkingService. Unable to establish connection");
+                return;
+            }
+            else
+            {
+                net.DisconnectAll();
+            }
+
+            net.OnConnected += OnConnected;
+            net.OnDisconnected += OnDisconnected;
+            net.OnReceive += OnReceive;
+
+            if (runAsUser)
+            {
+                net.StartListening(userPort);
+            }
+            else
+            {
+                net.ConnectTo(userAddress, userPort);
+            }
+        }
+
+        private void OnConnected(SocketEndpoint obj)
+        {
+            string playerId = string.Empty;
+            lock(playerIds)
+            {
+                if (!playerIds.ContainsKey(obj.Address))
+                {
+                    playerIds.Add(obj.Address, Guid.NewGuid().ToString());
+                }
+
+                playerId = playerIds[obj.Address];
+            }
+
+            if (playerId != string.Empty)
+            {
+                PlayerConnected?.Invoke(playerId);
+            }
+        }
+
+        private void OnDisconnected(SocketEndpoint obj)
+        {
+            string playerId = string.Empty;
+            lock (playerIds)
+            {
+                if (playerIds.ContainsKey(obj.Address))
+                {
+                    playerId = playerIds[obj.Address];
+                }
+            }
+
+            if (playerId != string.Empty)
+            {
+                PlayerDisconnected?.Invoke(playerId);
+            }
+        }
+
+        private void OnReceive(IncomingMessage obj)
+        {
+            bool validId = false;
+            string playerId = String.Empty;
+
+            lock(playerIds)
+            {
+                if (playerIds.ContainsKey(obj.Endpoint.Address))
+                {
+                    playerId = playerIds[obj.Endpoint.Address];
+                    validId = true;
+                }
+            }
+
+            if (validId)
+            {
+                DataReceived?.Invoke(playerId, obj.Data);
+            }
```

Comment:
Why does this function use a validId bool (but other ones above check against string.Empty)? 

Also, this uses String.Empty but things above use string.Empty.